packages/espressocash_client/lib/src/espressocash_client.dart

```diff
@@ -53,55 +74,44 @@ class EspressoCashClient {
 
   late final ClientChannel _channel;
   late UserServiceClient _userServiceClient;
-  late final RateServiceClient _rateServiceClient;
-  late final PaymentServiceClient _paymentServiceClient;
+  late RateServiceClient _rateServiceClient;
+  late PaymentServiceClient _paymentServiceClient;
   late ShortenerServiceClient _shortenerServiceClient;
-  late final ReferralServiceClient _referralClient;
-  late final dln_proto.DlnServiceClient _dlnServiceClient;
-  late final MoneygramServiceClient _moneygramServiceClient;
-  late final TokensServiceClient _tokensServiceClient;
+  late ReferralServiceClient _referralClient;
+  late dln_proto.DlnServiceClient _dlnServiceClient;
+  late MoneygramServiceClient _moneygramServiceClient;
+  late TokensServiceClient _tokensServiceClient;
 
-  static Future<EspressoCashClient> create({
-    String? baseUrl,
-    int? port,
-    required SignRequest sign,
-    String? walletAddress,
-    bool secure = true,
-  }) async {
-    final client = EspressoCashClient._(
-      baseUrl: baseUrl ?? defaultBaseUrl,
-      port: port,
-      sign: sign,
-      walletAddress: walletAddress,
-      secure: secure,
-    );
-    if (walletAddress != null && walletAddress.isNotEmpty) {
-      await client._init();
-    }
+  void initWithToken(String token) {
+    final callOptions = CallOptions(metadata: {'authorization': 'Bearer $token'});
 
-    return client;
+    _userServiceClient = UserServiceClient(_channel, options: callOptions);
+    _rateServiceClient = RateServiceClient(_channel, options: callOptions);
+    _paymentServiceClient = PaymentServiceClient(_channel, options: callOptions);
+    _shortenerServiceClient = ShortenerServiceClient(_channel, options: callOptions);
+    _referralClient = ReferralServiceClient(_channel, options: callOptions);
+    _dlnServiceClient = dln_proto.DlnServiceClient(_channel, options: callOptions);
+    _moneygramServiceClient = MoneygramServiceClient(_channel, options: callOptions);
+    _tokensServiceClient = TokensServiceClient(_channel, options: callOptions);
   }
 
   Future<void> dispose() async {
     await _channel.shutdown();
   }
 
-  Future<void> _init() async {
+  Future<String> login() async {
```

Comment:
Let's probably make `login` accept optional `token` parameter, and make `initWithToken` private method, to simplify the client's flow.

Reply:
updated login api